propagators/aws/xray/xrayidgenerator/aws_xray_idgenerator.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xrayidgenerator
+
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"encoding/hex"
+	"math/rand"
+	"strconv"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+type IDGenerator struct {
```

Comment:
This stutters: `xrayidgenerator.IDGenerator`.  If it can be moved into the `xray` directory, it would improve to `xray.IDGenerator`, but i'm not sure if that is feasible.